examples/groth16-verifier/src/lib.rs

```diff
@@ -0,0 +1,186 @@
+// Copyright 2023 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use anyhow::Result;
+use bonsai_sdk::alpha as bonsai_sdk;
+use byteorder::{BigEndian, WriteBytesExt};
+use hello_world_methods::{MULTIPLY_ELF, MULTIPLY_ID};
+use risc0_zkvm::{
+    default_prover, groth16::Groth16Seal, serde::to_vec, ExecutorEnv, Groth16Receipt, MemoryImage,
+    Program, Receipt, GUEST_MAX_MEM, PAGE_SIZE,
+};
+use std::time::Duration;
+
+// Run Bonsai for the hello world example
+pub fn run_bonsai(input_data: Vec<u8>) -> Result<Receipt> {
+    let client = bonsai_sdk::Client::from_env(risc0_zkvm::VERSION)?;
+
+    // create the memoryImg, upload it and return the imageId
+    let img_id = {
+        let program = Program::load_elf(MULTIPLY_ELF, GUEST_MAX_MEM as u32)?;
+        let image = MemoryImage::new(&program, PAGE_SIZE as u32)?;
+        let image_id = hex::encode(image.compute_id());
+        let image = bincode::serialize(&image).expect("Failed to serialize memory img");
+        client.upload_img(&image_id, image)?;
+        image_id
+    };
```

Comment:
This is another reason to drop Bonsai from this example, I think it will be a lot easier to just deal with zkvm first and get that all working then we can deal with integrating this into the `Prover` trait and then we can have an example that works for Bonsai or Local in just the same way that all other examples work, you just specify Bonsai env vars.